apps/frontend/core-dapp/src/stores/BalancerStore.ts

```diff
@@ -0,0 +1,131 @@
+import { BigNumber, utils } from 'ethers'
+import { SupportedNetworks } from 'prepo-constants'
+import { getContractAddress } from 'prepo-utils'
+import { makeAutoObservable, reaction, runInAction } from 'mobx'
+import { RootStore } from './RootStore'
+import { SupportedContracts } from '../lib/contract.types'
+import { BalancerQueriesAbi__factory } from '../../generated/typechain'
+import { supportedContracts } from '../lib/supported-contracts'
+
+const wethAddressByNetowrk: Partial<Record<SupportedNetworks, string>> = {
+  arbitrumOne: '0x82aF49447D8a07e3bd95BD0d56f35241523fBab1',
+}
+
+const WETH_DECIMALS = 18
+
+const wstEthWethBalancerPoolIdByNetwork: Partial<Record<SupportedNetworks, string>> = {
+  arbitrumOne: '0x36bf227d6bac96e2ab1ebb5492ecec69c691943f000200000000000000000316',
+}
+
+export class BalancerStore {
+  private readonly balancerQueriesInterface = BalancerQueriesAbi__factory.createInterface()
+  private readonly root: RootStore
+  private wstEthPriceInEthBN: BigNumber | undefined = undefined
```

Comment:
Should we keep this accessible to other stores for calculations?
```suggestion
  wstEthPriceInEthBN: BigNumber | undefined = undefined
```

Reply:
I would rather keep everything that we don't need to expose *right now* private, and then refactor if we do need to access it outside stores.

Also I wish Javascript had an easy way of doing `private set` :/